lib/utils.js

```diff
@@ -417,6 +417,18 @@ function removeTicks(s, tickChar) {
 }
 exports.removeTicks = removeTicks;
 
+function flattenObject(obj, flattenedObj, subPath) {
```

Comment:
rather than mutating `flattenedObj` it just simple create one internally and return it. Any specific reasoning for opting this behavior

Reply:
I changed the function so as to instantiate the returned value and return it when the flattening process is done.